clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -1399,8 +1400,10 @@ mlir::LogicalResult CIRToLLVMShiftOpLowering::matchAndRewrite(
   if (op.getIsShiftleft()) {
     rewriter.replaceOpWithNewOp<mlir::LLVM::ShlOp>(op, llvmTy, val, amt);
   } else {
-    assert(!cir::MissingFeatures::vectorType());
-    bool isUnsigned = !cirValTy.isSigned();
+    const bool isUnsigned =
+        cirValTy
+            ? !cirValTy.isSigned()
+            : !mlir::cast<cir::IntType>(cirValVTy.getElementType()).isSigned();
```

Comment:
Would it make sense for the vector type to support `isSigned()`? The instead of having separate cirValTy and cirValVTy (one of which would always be null), we could do something like this above:

```
if (!cirValTy) {
  cirValTy = mlir::cast<cir::VectorType>(op.getValue().getType());
  assert(mlir::isa<cir::VectorType>(op.getAmount().getType());
}
```

Reply:
I think we can also simplify the asserts, for example

```
assert((cirValTy == op.getType() || cirValVTy == op.getType()) &&
         "inconsistent operands' types NYI");
```

to this, so we check that the value type equals to `op.type`

```
assert((op.getValue().getType() == op.getType()) &&
         "inconsistent operands' types NYI");
```

and 

```
assert(((cirValTy && cirAmtTy) || (cirAmtVTy && cirValVTy)) &&
         "shift input type must be integer or vector type, otherwise NYI");
```

To just check that both are true or false, so both are ints or vectors 🤔 

```
 assert((cirValTy == cirAmtTy) &&
         "shift input type must be integer or vector type, otherwise NYI");
```

Reply:
I used your suggestion and assigned isUnsigned and getLLVMIntCast in the same branch